File: nomad-samples/python/Guest Registration/account/register_guest.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json
import requests

# @param {string} AUTH_TOKEN - The authorization token

def register_guest(AUTH_TOKEN: str) -> dict:
    if not AUTH_TOKEN:
        raise Exception("Authorization token not found")
  
    API_URL = "https://" + PORTAL_API_URL + "/account/register-guest"

    HEADERS = {
        "Content-Type": "application/json",
      	"Authorization": "Bearer " + AUTH_TOKEN
    }
    
    # replace username and password with your username and password
    BODY = {
      	"email": "pawev59721@ozatvn.com",
        "firstName": "Guest",
        "lastName": "User",
        "organization": "Nomad Media, Corp",
        "password": "nepqub-nuJvo1-woczih!",
        "passwordConfirmation": "nepqub-nuJvo1-woczih!"
    }

    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        if RESPONSE.status_code != 200:
            raise Exception("Response returned " + str(RESPONSE.status_code))

        return json.loads(RESPONSE.text)
    except:
      	api_exception_handler(RESPONSE, "Registering guest failed")

File: nomad-samples/portal/js/guest-registration/guest/register-guest.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js"; 

export default async function registerGuest(AUTH_TOKEN, EMAIL, PASSWORD) 
{
		// Create header for the request
    const HEADERS = new Headers(AUTH_TOKEN);
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
  	const BODY = {
        email: EMAIL,
        firstName: "Guest",
        lastName: "User",
        organization: "Nomad Media, Corp",
        password: PASSWORD,
        passwordConfirmation: PASSWORD
    };
  
  // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/account/register-guest`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Register guest failed");
}